packages/grafana-data/src/transformations/transformers/calculateField.ts

```diff
@@ -98,10 +103,18 @@ export const calculateFieldTransformer: DataTransformerInfo<CalculateFieldTransf
           creator = getBinaryCreator(defaults(binaryOptions, defaultBinaryOptions), data);
         } else if (mode === CalculateFieldMode.Index) {
           return data.map((frame) => {
+            const indexArr = [...Array(frame.length).keys()];
+
+            if (options.index?.showPercentage) {
+              for (let i = 0; i < indexArr.length; i++) {
+                indexArr[i] = (indexArr[i] / indexArr.length) * 100;
```

Comment:
I recently discovered that there's a **Standard options > Unit** of **Percent (0.0-1.0)**

Users can lean on this **Unit** whenever they need to render percentages from 0% to 100%. I can imagine other users might not want to use a scale of percent, but some other quantile.

So how about we don't multiple by 100 at all, and instead of restricting this transform by naming it as "percentage" we could make this more generic and call it something like "quantile from row index"?

```suggestion
                indexArr[i] = (indexArr[i] / indexArr.length);
```

Reply:
yeah, i think leaving it without * 100 and then using unit is probably better 👍